### PR TITLE
[ProjectSearch] toggle between project text search and file name search

### PR DESCRIPTION
**Pull request by aklt**
The code is not too pretty, but it does what is required and since this
is a temporary ui I think it is ok.

Autocomplete was modified to render children before results

Associated Issue: #3091 
### Summary of Changes

* This adds UI for the project text search and files search (CMD or CTRL + o and CMD or CTRL + Shift + f) as shown in the image in issue 3091 
* This is temporary UI

### Test Plan

* Enable "projectTextSearch" in `configs/local.json`
* Press the above key combos to bring up the search bar
* Click on the active "files" or "text" link to change the UI

### Screenshots

toggle-search-1

### CODE REVIEW

File: src/components/ProjectSearch/index.js [hunk 1]
```diff
@@ -118,12 +100,27 @@ class ProjectSearch extends Component {
...
+    });
   }
 
   renderTextSearch() {
     const { sources, closeActiveSearch } = this.props;
-    return TextSearch({ sources, closeActiveSearch });
+    return TextSearch({
+      sources,
+      closeActiveSearch,
+      searchBottomBar: ToggleSearch({
```
Comment (jasonLaster) [inline src/components/ProjectSearch/index.js]: smart 👍

File: src/components/ProjectSearch/index.js [hunk 2]
```diff
@@ -118,12 +100,27 @@ class ProjectSearch extends Component {
 
   renderSourceSearch() {
     const { sources, selectSource, closeActiveSearch } = this.props;
-    return SourceSearch({ sources, selectSource, closeActiveSearch });
+    return SourceSearch({
+      sources,
+      selectSource,
+      closeActiveSearch,
+      searchBottomBar: ToggleSearch({
+        kind: "files",
```
Comment (jasonLaster) [inline src/components/ProjectSearch/index.js]: i think we should stay consistent with the search type and call it "sources"

File: assets/panel/debugger.properties [hunk 3]
```diff
@@ -466,7 +466,7 @@ welcome.search=%S to search for sources
 
 # LOCALIZATION NOTE (sourceSearch.search): The center pane Source Search
 # prompt for searching for files.
-sourceSearch.search=Search Sources…
```
Comment (jasonLaster) [inline assets/panel/debugger.properties]: hmm - what do others think I like this new name.

If we change it we should bump the key to `sourceSearch.search2`

Comment (aklt) [inline assets/panel/debugger.properties]: I was not sure If I should change it, but i think i like it better

File: src/components/ProjectSearch/index.js (hunk 2 above)
Comment (aklt) [inline src/components/ProjectSearch/index.js]: Ok, I will change that.  Screenshot coming up

**Comment by jasonLaster** [comment]
Looks like the styling could be tweaked:

1. can you use the selected blue we use in other places
2. search is too bold
3. hover states should have a pointer cursor and slight grey
4. it looks like we're highlighting the wrong one (clicking behavior) is not what i'd expect

**Comment by jasonLaster** [comment]
Perhaps we want to change the style of the links.

so it's two options: "File Files", "Find in Project"

**Comment by aklt** [comment]
In which way should the style be changed?
I changed the blue text to be just spans btw.

**Comment by jasonLaster** [comment]
Perhaps just two links. Sorry it's tough to describe. I dont mind jumping
in

**Comment by aklt** [comment]
not sure I understand, you are welcome to jump in of course ;-)

**Comment by jasonLaster** [comment]
@aklt I made some tweaks to the look and feel and the source code. 

I really like how bottom bar was structured. I updated the ternary a bit so we wouldn't duplicate logic

**Comment by aklt** [comment]
@jasonLaster Looks good to me. It is good to have less dom probably and you can also toggle the link if it is active as far as I can see.  I can try it out tomorrow evening :-)

**Pull request merged**